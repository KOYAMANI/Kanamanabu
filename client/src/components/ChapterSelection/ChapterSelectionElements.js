import styled from 'styled-components';


export const ChapterSelectionContainer = styled.div`
  background: orange;
  height: 500px;
  width: 400px;

`;