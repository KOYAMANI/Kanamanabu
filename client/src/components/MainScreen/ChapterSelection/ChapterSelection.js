import React from 'react'
import { Col } from 'react-bootstrap'
import { 
    ChapterSelectionTitle,
    ChapterSelectionSubTitle,
    ChapterSelectionBox,
} from './ChapterSelectionElements';


const ChapterSelection = ({category}) => {
    return (
        <ChapterSelectionBox background={ Object.values(category.imageUrl) }>
            <Col>
                <ChapterSelectionTitle>
                    {category.category}
                </ChapterSelectionTitle>
                <ChapterSelectionSubTitle>
                    {category.title}
                </ChapterSelectionSubTitle>
            </Col>
        </ChapterSelectionBox>
    )
}

export default ChapterSelection
