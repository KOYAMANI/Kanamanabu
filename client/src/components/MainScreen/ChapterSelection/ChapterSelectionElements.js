
import styled from 'styled-components';
import { Box } from '@material-ui/core';

export const ChapterSelectionBox = styled(Box)`
    margin-right: 10px;
    width: 256px;
    height: 450px;
    background-color: #696969;
    display: flex;
    align-items: flex-end;
    background-image: url(${props => props.background}) ;
    background-size: cover;
    &:hover {
        border :4px solid #fff;
        background-color: #808080;
      }
    @media (max-width: 1200px) {
    width: 170px;  
    height: 350px;
    }
`

export const ChapterSelectionTitle = styled.p`
    color: #ffffff;
    font-size: 24px;
    font-weight: bold;
  @media (max-width: 800px) {
    font-size: 20px;
  }
`

export const ChapterSelectionSubTitle = styled.p`
    color: #ffffff;
    font-size: 18px;
  @media (max-width: 800px) {
    font-size: 14px;
  }
`