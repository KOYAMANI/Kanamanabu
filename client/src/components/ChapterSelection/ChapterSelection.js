import React from 'react';
import { ChapterSelectionContainer } from "./ChapterSelectionElements";
import { useSelector } from "react-redux";

const ChapterSelection =()=> {

    const chapter = useSelector(state => state.currentChapter);
    const {currentChapter} = chapter;

    return (
            <ChapterSelectionContainer>

                <div>{currentChapter?
                    (currentChapter[0].title === 'hiragana'
                    ||currentChapter[0].title === 'katakana')?
                    <div>
                        {currentChapter.map(selection =>
                            <div key={selection.consonant}>
                                <span className="styles.capital" >
                                    {selection.consonant}
                                </span>
                                <span>{selection.letters}</span>
                            </div>
                        )}
                    </div>
                        :<p>{currentChapter[0].title}</p>
                    : 'no selection'
                }</div>
           < / ChapterSelectionContainer>
        );
}


export default ChapterSelection;