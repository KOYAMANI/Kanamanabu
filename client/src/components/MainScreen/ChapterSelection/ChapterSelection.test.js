import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer'
import ChapterSelection from "./ChapterSelection";

afterEach(cleanup)

const category =
    {
        _id: "1",
        title: 'title-test',
        category: 'subTitle-test',
        subcategory: ['aiueo', 'kakikukeko'],
        desc: 'desc',
        imageUrl: 'https://google.com'
    }

describe('ChapterSelection', ()=>{
    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<ChapterSelection category={category}/>, div)
    });

    test('Render Title', ()=>{
        render(<ChapterSelection category={category}/>)
        const title = screen.getByText('title-test')
        expect(title).toBeInTheDocument();
    })

    test('Render SubTitle', ()=>{
        render(<ChapterSelection category={category}/>)
        const subTitle = screen.getByText('subTitle-test')
        expect(subTitle).toBeInTheDocument();
    });

    test('match snapshot',()=>{
        const tree = renderer.create(<ChapterSelection category={category}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

})
