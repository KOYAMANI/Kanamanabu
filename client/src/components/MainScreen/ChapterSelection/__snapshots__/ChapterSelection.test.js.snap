// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterSelection match snapshot 1`] = `
<div
  background={
    Array [
      "h",
      "t",
      "t",
      "p",
      "s",
      ":",
      "/",
      "/",
      "g",
      "o",
      "o",
      "g",
      "l",
      "e",
      ".",
      "c",
      "o",
      "m",
    ]
  }
  className="MuiBox-root MuiBox-root-4 sc-bczRLJ ihOJYD"
>
  <div
    className="col"
  >
    <p
      className="sc-gsnTZi isGCWB"
    >
      subTitle-test
    </p>
    <p
      className="sc-dkzDqf dOftdm"
    >
      title-test
    </p>
  </div>
</div>
`;
